web/api/v1/api.go

```diff
@@ -1479,6 +1443,18 @@ func (api *API) respondError(w http.ResponseWriter, apiErr *apiError, data inter
 	}
 }
 
+func parseTimeParam(r *http.Request, paramName string, defaultValue time.Time) (int64, time.Time, error) {
```

Comment:
I think I'd prefer keeping the return signature more "normalized" (less redundant) and simpler by just returning `time.Time` and letting the user do the `timestamp.FromTime()` whenever it is really needed.

Reply:
I commented for this in the original PR, but I don't mind either way.

Reply:
Yep, this matches with @roidelapluie comment https://github.com/prometheus/prometheus/pull/6860#discussion_r383118014